collector/src/main/java/com/bitmovin/analytics/BitmovinAdAnalytics.kt

```diff
@@ -0,0 +1,225 @@
+package com.bitmovin.analytics
+
+import com.bitmovin.analytics.ads.*
+import com.bitmovin.analytics.data.AdSample
+import com.bitmovin.analytics.data.AdEventData
+import com.bitmovin.analytics.utils.Util
+
+
+class BitmovinAdAnalytics(var analytics: BitmovinAnalytics) {
+    private var activeAdBreak: AdBreak? = null
+    private var activeAdSample: AdSample? = null
+    private var adPodPosition: Int = 0
+    private var adStartupTimestamp: Long? = null
+    private var currentTime: Long? = null
+    private var beginPlayingTimestamp: Long? = null
+    private var isPlaying: Boolean = false
+    private val adManifestDownloadTimes: HashMap<String, Long?> = hashMapOf()
+
+    fun onAdStarted(ad: Ad) {
+        if (ad.isLinear != true) {
+            return
+        }
+
+        this.resetActiveAd()
+        this.activeAdSample = AdSample(ad = ad)
+        this.currentTime = null
+        this.activeAdSample!!.adStartupTime = if (this.adStartupTimestamp != null) Util.getTimeStamp() - this.adStartupTimestamp!! else null
```

Comment:
`!!` will throw NPE if not set. We should guard against it

Reply:
two lines above there is a constructor call 
should be enough right?